core/src/test/kotlin/org/evomaster/core/problem/rest/RestCallResultTest.kt

```diff
@@ -0,0 +1,29 @@
+package org.evomaster.core.problem.rest
+
+import org.junit.Assert.assertEquals
+import org.junit.jupiter.api.Test
+import javax.ws.rs.core.MediaType
+
+internal class RestCallResultTest {
+
+    @Test
+    fun givenAStringIdWhenGetResourceIdThenItIsReturnedAsString() {
+        val rc = RestCallResult(false)
+        rc.setBody("{\"id\":\"735\"}")
+        rc.setBodyType(MediaType.APPLICATION_JSON_TYPE)
+
+        val res = rc.getResourceId()
+
+        assertEquals("735", res)
+    }
+    @Test
+    fun givenANumericIdWhenGetResourceIdThenItIsReturnedAsString() {
+        val rc = RestCallResult(false)
+        rc.setBody("{\"id\":735}")
+        rc.setBodyType(MediaType.APPLICATION_JSON_TYPE)
+
+        val res = rc.getResourceId()
+
+        assertEquals("735", res)
+    }
```

Comment:
nice tests!
indeed, whether the id is a number or a string, we should not quote it with " in the URI paths